CONTRIBUTING.md

```diff
@@ -18,45 +18,163 @@ any additional information for contributing code to this project.
+
+macOS:
+
 ```
 $ source venv/bin/activate
 ```
 
-- Install requirements:
+Windows:
+
+```
+venv\Scripts\activate.bat
+```
+
+3. Install requirements
+
 ```
 $ pip3 install -r requirements.txt
 ```
 
-You can now run the tests and the CLI with modifiable files!
+4. You can now run the tests and the CLI with modifiable files.
+
+Check it out! Run the following prefix to start seeing changes
+
+```
+$ ./pkg_bin/conjur <command> <subcommand>
+```
 
 ## Testing
 
 ### Unit and Integration tests
 
+To run both unit and integration tests at once in a containerized environment:
+
 ```
 $ ./bin/test
 ```
 
+To run unit tests:
+```
+./bin/test_unit
+```
+
+You can run integration tests in the following ways:
+1. In a containerized environment (Python required)
+
+1. As a process (no Python required)
+
+#### Running in a containerized environment
+
+To run specific tests, perform the following:
+
+1. Drop down into a test container
+
+```
+$ ./bin/test_integration -d
+```
+
+1. Under the individual test add the following:
+
+```
+<name-of-test-function>.<identifier>=True
+
+## Example
+def my-integration-test-function()
+  ...
+
+my-integration-test-function.someidentifier=True
+```
+
+1. Add the identifier of choice to the following command:
+
+```
+root@123456:/opt/conjur-api-python3# nose2 -v -X --config integration_test.cfg -A '<identifier>' $@
+
+## Example
+root@123456:/opt/conjur-api-python3# nose2 -v -X --config integration_test.cfg -A 'someidentifier' $@
+```
+
+1. You should see that only that specific test is run. Every change made locally can be seen in the container so you
+do *not* need to rebuild before running these tests again.
+
+#### Running as a process
+
+We provide the option to run the integration tests as a process. This means integration tests can run in the same way
+a user would use our CLI; without Python installed on the machine, using the Conjur CLI exec.
+
+The integration tests are wrapped in an integration_test_runner Python module to run in a Python-free environment.
+That way, tests can be run cross-platform.
+
+##### Setup
+
+1. Pack the `integration_test_runner.py` using PyInstaller in the platform to run the executable.
+
+  To pack: `pyinstaller -F test/util/test_runners/integration_test_runner.py`. Note that you will need
+  to pack each runner in each platform that you want to run the tests.
+
+1. Pack the Conjur CLI using PyInstaller in the platform to run the executable
+
+  To pack: `pyinstaller -F ./pkg_bin/conjur`. Note that you will need to pack each runner in each platform
+  that you want to run the tests. Also note that _only_ for macOS, you should pack the CLI as a directory instead of a
+  single file (using -D instead of -F). This will allow the CLI tests to run quicker.
+
+1. Run the executable: `./integration_test_runner`, supplying the below required parameters via the command line.
+
+###### Required parameters
+
+`--invoke_cli_as_process`, required to run the CLI as a process.
```

Comment:
i am used to flags being separated by `-` instead of `_`. Is this intentional?

Reply:
i would change `,` to ` - ` or `:`

Reply:
will change